pytorch_lightning/utilities/cli.py

```diff
@@ -396,16 +395,21 @@ def parse_arguments(self, parser: LightningArgumentParser) -> None:
     def before_instantiate_classes(self) -> None:
         """Implement to run some code before instantiating the classes."""
 
-    def instantiate_classes(self, subcommand: Optional[str]) -> None:
+    def instantiate_classes(self) -> None:
         """Instantiates the classes and sets their attributes."""
         self.config_init = self.parser.instantiate_classes(self.config)
         self.datamodule = self._get(self.config_init, "data")
         self.model = self._get(self.config_init, "model")
-        callbacks = [self._get(self.config_init, c) for c in self._parser(subcommand).callback_keys]
-        self.trainer = self.instantiate_trainer(self._get(self.config_init, "trainer"), callbacks)
+        self._add_configure_optimizers_method_to_model(self.subcommand)
+        self.trainer = self.instantiate_trainer()
 
-    def instantiate_trainer(self, config: Dict[str, Any], callbacks: List[Callback]) -> Trainer:
+    def instantiate_trainer(self, callbacks: Optional[List[Callback]] = None) -> Trainer:
```

Comment:
Should we enable any parameters to be passed as *args, **kwargs. This might be useful for loggers / changing the max_epochs, etc...

Reply:
Created here: https://github.com/PyTorchLightning/pytorch-lightning/issues/9247

Reply:
Good idea! I'll change it